### PR TITLE
Switch to nf-validation

### PR DESCRIPTION
**Pull request by jonasscheid**
Intitially wanted to fix an issue in the samplesheet checker with `.vcf.gz`, but then I realized we should move to nf-validation to get rid of some custom code and properly check parameters.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/epitopeprediction _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jonasscheid** [comment]
@christopher-mohr Do you know if if anyone still uses the variant tsv format or GSvar? Can we deprecate it? Does not seem to be a community-format 🤔 or am I mistaken

**Comment by christopher-mohr** [comment]
> community-format

How do you define `community-format`? In first place this was implemented since we usually received the data in this format. Not sure what is used at the moment at QBiC etc. However, in general this should not be part of this PR.

**Comment by jonasscheid** [comment]
> How do you define community-format? 

A standardised text file format that represents information about SNPs, indels and structural variants that the majority of scientists use.

> Not sure what is used at the moment at QBiC etc.

Not used as people told me

> However, in general this should not be part of this PR.

True, but since it is related to the errors, I was questioning whether these formats will be maintained and whether we should support them. Otherwise, I would set up another PR deprecating the related tests and functionality

**Comment by christopher-mohr** [comment]
> True, but since it is related to the errors, I was questioning whether these formats will be maintained and whether we should support them. Otherwise, I would set up another PR deprecating the related tests and functionality

The problem seems to be that you are providing a tsv file with variants as peptide input.

**Comment by jonasscheid** [comment]
That is why I asked regarding maintaining these variant tsv is still a thing we want to do. 

> Otherwise, I would set up another PR deprecating the related tests and functionality

of variant tsv and GSvar

**Comment by christopher-mohr** [comment]
> That is why I asked regarding maintaining these variant tsv is still a thing we want to do.
> 
> > Otherwise, I would set up another PR deprecating the related tests and functionality
> 
> of variant tsv and GSvar

 Yes I know. :) We should just be sure that it's not a problem in general with the added changes.

File: workflows/epitopeprediction.nf [hunk 1]
```diff
@@ -94,43 +90,59 @@ import groovy.json.JsonSlurper
...
-                            return [ meta_data, input_file ]
-                        variant_uncompressed :  meta_data.inputtype == 'variant'
-                            return [ meta_data, input_file ]
-                        peptide :  meta_data.inputtype == 'peptide'
-                            return [ meta_data, input_file ]
-                        protein :  meta_data.inputtype == 'protein'
-                            return [ meta_data, input_file ]
-                    }
-                .set { ch_samples_from_sheet }
+    // Function to read the alleles from a file or stage it from url
```
Comment (christopher-mohr) [inline workflows/epitopeprediction.nf:113]:
```suggestion
    // Function to read the alleles from a file or use given string
```

File: workflows/epitopeprediction.nf [hunk 2]
```diff
@@ -94,43 +90,59 @@ import groovy.json.JsonSlurper
...
+
+    ch_input = Channel.fromSamplesheet("input")
+    ch_input
+        .branch {
+            sample, alleles, mhc_class, filename ->
+                def allele_list = readAlleles(alleles)
+                validate_alleles(allele_list, mhc_class)
+                variant_compressed : filename.endsWith('.vcf.gz')
+                    return [[sample:sample, alleles:allele_list, mhc_class:mhc_class, inputtype:'variant_compressed'], filename ]
+                variant_uncompressed : filename.endsWith('.vcf') || filename.endsWith('.GSvar')
```
Comment (christopher-mohr) [inline workflows/epitopeprediction.nf:133]:
```suggestion
                variant_uncompressed : filename.endsWith('.vcf')
```

File: workflows/epitopeprediction.nf [hunk 3]
```diff
@@ -94,43 +90,59 @@ import groovy.json.JsonSlurper
...
+                protein : filename.endsWith('.fasta') || filename.endsWith('.fa')
+                    return [[sample:sample, alleles:allele_list, mhc_class:mhc_class, inputtype:'protein'], filename ]
+    } .set { ch_samples_from_sheet }
 
     // gunzip variant files
     GUNZIP_VCF (
         ch_samples_from_sheet.variant_compressed
     )
-    ch_versions = ch_versions.mix(GUNZIP_VCF.out.versions.ifEmpty(null))
+    ch_versions = ch_versions.mix(GUNZIP_VCF.out.versions)
```
Comment (christopher-mohr) [inline workflows/epitopeprediction.nf:145]: Are we sure that this is not needed anymore?

**Review by christopher-mohr — COMMENTED** [review]
Just some minor things

File: workflows/epitopeprediction.nf (hunk 3 above)
Comment (jonasscheid) [reply workflows/epitopeprediction.nf:145]: Not needed anymore for versions channel. In fact will throw a warning with next tools update: https://github.com/nf-core/tools/issues/2506

**Review by christopher-mohr — APPROVED** [review]
LGTM

**Pull request merged**